key-rotator/manifest/manifest.go

```diff
@@ -36,12 +37,16 @@ type DataShareProcessorSpecificManifest struct {
 	PacketEncryptionKeyCSRs PacketEncryptionKeyCSRs `json:"packet-encryption-keys"`
 }
 
-func (m DataShareProcessorSpecificManifest) Equal(o DataShareProcessorSpecificManifest) bool {
+func (m DataShareProcessorSpecificManifest) equalModuloKeys(o DataShareProcessorSpecificManifest) bool {
 	return m.Format == o.Format &&
 		m.IngestionIdentity == o.IngestionIdentity &&
 		m.IngestionBucket == o.IngestionBucket &&
 		m.PeerValidationIdentity == o.PeerValidationIdentity &&
-		m.PeerValidationBucket == o.PeerValidationBucket &&
+		m.PeerValidationBucket == o.PeerValidationBucket
+}
+
+func (m DataShareProcessorSpecificManifest) Equal(o DataShareProcessorSpecificManifest) bool {
```

Comment:
nit: Just keep in mind that this is a pretty large struct that we're going to be copying each time we want to call Equal. Not a big deal and premature optimization at best.

Reply:
Yup -- I like to pass things by-value, until there's a good reason not to (e.g. performance, necessary mutability, ...). IMO, it's somewhat less error-prone to use values rather than pointers-to-values because you can be sure passing your value to some function won't mutate it; though admittedly in this case, there are several map fields in the struct so this goal is not fully realized. (and keep in mind that what is copied for e.g. a string is the pointer-and-length, not the underlying byte array, so these copies are of 11 64-bit words per struct copy, rather than an arbitrary amount of data)